src/main/scala/scala/tools/refactoring/analysis/SymbolExpanders.scala

```diff
@@ -205,4 +207,37 @@ trait DependentSymbolExpanders extends TracingImpl {
       case _ => Nil
     })
   }
+
+  /**
+   * Associates term symbols with missing ranges to related symbols that have ranges.
+   *
+   * The reason that we need this is that in some cases, the PC generates multiple
+   * symbols for one and the same symbol in user source code, one of them with a
+   * proper range position, and others just with offset positions. One place where
+   * this happens is in desugared for comprehensions with filter clauses.
+   * See Assembler Ticket #1002650.
+   */
+  trait TermsWithMissingRanges extends SymbolExpander { this: IndexLookup =>
+    protected abstract override def doExpand(s: Symbol): List[Symbol] = {
+      termsWithSamePointButRange(s) ::: super.doExpand(s)
+    }
+
+    private def termsWithSamePointButRange(s: Symbol): List[Symbol] = {
+      val lookAtSymbol = {
+        s != NoSymbol && !s.isSynthetic && s.isInstanceOf[TermSymbol] &&
+          s.pos.isInstanceOf[OffsetPosition] && !s.pos.isInstanceOf[RangePosition]
```

Comment:
just a question why `s.pos.isRange` and `s.isTerm` are too weak for this check. Or maybe it is not this position and symbol?